2009-02-13  Michael Tautschnig  <mt@debian.org>

	* bin/fai-setup: Do not unconditionally add $FAI_CONFIGDIR to exports(5)
		(thanks Sebastian Harl) (closes: #504801)
Index: trunk/bin/fai-setup
===================================================================
--- trunk.orig/bin/fai-setup
+++ trunk/bin/fai-setup	
@@ -178,7 +178,10 @@
 
     addr=$(LC_ALL=C ifconfig $SERVERINTERFACE | grep -v 127.0.0.1 | perl -ne '/addr:([\d.]+)/ && print"$1\n"'|head -1)
     mask=$(LC_ALL=C ifconfig $SERVERINTERFACE | grep -v 127.0.0.1 | perl -ne '/Mask:([\d.]+)/ && print"$1\n"'|head -1)
-    add_export_line $FAI_CONFIGDIR  "$addr/$mask(async,ro,no_subtree_check)"
+
+    if expr match "$FAI_CONFIG_SRC" 'nfs:\/\/' > /dev/null; then
+        add_export_line $FAI_CONFIGDIR  "$addr/$mask(async,ro,no_subtree_check)"
+    fi
     add_export_line $NFSROOT  "$addr/$mask(async,ro,no_subtree_check,no_root_squash)"
     if [ -z "$nfsserver" ]; then
 	echo "Could not find the type of your nfs server. Maybe"
